bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details/ConicNameEditor.scala

```diff
@@ -41,6 +41,10 @@ final class ConicNameEditor(date: HorizonsIO[Date]) extends JPanel with Telescop
       override def textBoxKeyPress(tbwe: TextBoxWidget): Unit =
         nonreentrant {
           spt.getTarget.setName(tbwe.getValue)
+          // editing the name so invalidate the horizons id.
```

Comment:
Yeah it's a hack that says "I'm pushing model changes into the controls (or pushing control data into the model) so ignore any feedback look that might result". As Shane said it's easier than detaching and reattaching all the listeners.